rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,501 @@
+    
+### Refactoring note
+    
+**Observing**: From a refactoring perspective, it might also be interesting to see how many conflicts the different formats would cause.
+    
+nixpkgs comments:
+
+- `##` ~4k usages (most of them for visual separation, e.g., `###########`)
+- `#` ~20k usages
+- `/*` ~6k usages
+- `/**` 160 usages (most empty ?)
+    
+Choosing `/**` or subsets would cause minor conflicts within current nixpkgs. While this is NOT the main reason for the final decision, it MUST be considered.
+
+
+## Just free text as a content format
+
+While this allows the most freedom, it is usually considered the best option, not creating any restrictions. 
+
+However, nix/rfc72 defines commonMark as the official documentation format. 
+This is why we decided to follow this convention.
+
+## Consequences of not implementing this
+
+- By not implementing this feature, Nix gains no ability for tool-generated documentation.
+- Documentation will be defined by nixdoc, not by the nix community.
+- Many existing comments written for documentation will remain imperceptible.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Will `nix` itself implement native support like in rust -> `cargo doc`?
+
+- How can a tool keep the connection from where a docstring was defined and where the attribute was exposed (lib/default. Nix exposes mapAttrs which is defined at lib/attrsets. Nix)
+  - There are more complicated things.
+
+**Possible answer**: This is a tooling question; implementation details will not be discussed in detail. It is possible to track comments in the AST. As shown before in this PR: [nix/#1652](https://github.com/NixOS/nix/pull/1652)
+
+# Future work
+[Future]: #future-work
+
+## Editor support
+
+- Implement displaying the related documentation when hovering over an expression. (lspAction/hover)
+
+Nix already offers a bunch of LSPs, e.g., [nil](https://github.com/oxalica/nil), [rnix-lsp](https://github.com/nix-community/rnix-lsp) are the most common ones.
+
+
+## Nixdoc
+
+Nixdoc must be changed to differentiate between regular comments and doc-comments.
+There might be an intermediate phase of transition, where the old Syntax and features are supported in parallel to allow a phase of transition and refactoring of existing documentation comments.
+
+## Documentation generators
+
+- Future nixdoc could have a static `map.json` that contains the scope for every discovery file/path in nixpkgs.
+
+As this second approach is much easier, we propose that we initially start to extend the scope.
+
+Generating static documentation from doc-comments would still be excellent but remains a challenge if generated from Nix dynamically.
+If we want the fully automated approach, we need something to evaluate nix expressions. 
+
+We solve such concerns in `tvix` or in `nix`, which could provide a `nix-analyzer` that pre-evaluates expressions and their respective documentation. 
```

Comment:
`tvix` is not self-describing name and there are no more mentions of it in this doc. Same goes with `nix-analyzer`. I'm not sure I understand this sentence correctly, but maybe something like:

```suggestion
We solve such concerns in [`tvix`](https://github.com/tvlfyi/tvix) or in `nix`, which could vend a tool that pre-evaluates expressions and gathers their respective documentation. 
```

P.S. I tried to search for `nix-analyzer` and found only empty crate https://crates.io/crates/nix-analyzer

Reply:
The link is just an export for GitHub, https://tvix.dev/ is the website.

Reply:
@ony: Thanks, you caught a blind spot here. 

- One might not know tvix (see @flokli's link) 

- with `nix-analyzer`, I initially meant a fictional tool that we could build similar to `rust-analyzer`. It does not exist yet.

I updated the sentence